### ISSUE TITLE
ENOENT when a file with stashed information is deleted

### ISSUE THREAD

**Issue by garyb** [natefaubion/purescript-psa#17]
This is obviously not the usual way one would use `purescript-psa`, but if you have it installed as a local dependency and then try to compile the project with pulp it seems to be resolving the paths incorrectly:

```
fs.js:892
  return binding.stat(pathModule._makeLong(path));
                 ^

Error: ENOENT: no such file or directory, stat 'D:\Home\GitHub\slamdata\bower_components\purescript-markdown-halogen\src\Text\Markdown\SlamDown\Html.purs'
    at Error (native)
    at Object.fs.statSync (fs.js:892:18)
    at Object.stat (D:\Home\GitHub\slamdata\node_modules\purescript-psa\output\Node.FS.Sync\index.js:133:94)
    at __do (D:\Home\GitHub\slamdata\node_modules\purescript-psa\output\Main\index.js:510:61)
```

I'm assuming it's psa's fault here, but I guess it could equally be pulp, or some interaction of the two that is causing this. Thought I'd bring it up anyway, as it would be nice to be able to use psa this way so we can use `--censor-lib --strict` as part of Travis builds.

**Comment by garyb** [comment]
Uhh, I should have read the error message more carefully...

This is nothing to do with local dependencies, I think it occurs when a that there is stashed information for is deleted.

**Comment by natefaubion** [comment]
Hmmm, this case is supposed to be handled https://github.com/natefaubion/purescript-psa/blob/54c9065a3d6f174f1f6afe20a7d0532d0a47c7c7/bin/src/Main.purs#L307-L310 I have to call `stat` on all stashed references to check the timestamp, and if the file doesn't exist, the reference should be filtered out completely. Line 510 of `Main/index.js` is definitely that block of code though, so I don't know why `catchException` is failing.

**Comment by natefaubion** [comment]
Ah, this is just a case where dependencies need to be updated https://github.com/purescript-node/purescript-node-fs/commit/b1a99f855a245e1031a91e33ff218b5531c0033e

**Comment by natefaubion** [comment]
This is fixed in 0.3.6

**Comment by garyb** [comment]
Thanks!
